src/OVAL/probes/independent/xmlfilecontent.c

```diff
@@ -128,12 +128,18 @@ static int process_file(const char *path, const char *filename, void *arg)
 		whole_path[path_len] = FILE_SEPARATOR;
 		++path_len;
 	}
+	dI("%s", path);
 
 	memcpy(whole_path + path_len, filename, filename_len + 1);
 
-	/* evaluate xpath */
+	if (prefix == NULL) {
+		doc = xmlParseFile(whole_path);
+	} else {
+		char *path_with_prefix = oscap_sprintf("%s%s", prefix, whole_path);
```

Comment:
comment why you don't have to insert PATH_SEPARATOR in between. this is not obvious

Reply:
I don't know, the `whole_path` should come from `FTSENT fts_path`, which is described in `man 3 fts` as:

> The  path  for  the file relative to the root of the traversal.  This path contains the path specified to fts_open() as a prefix. 

but in  my cases it was not a relative path, it started with `/`.

How would you understand that?

Reply:
Your reply is precisely why I am asking about this. To write robust code you need to be sure about this or insert logic that makes sure about this.

Reply:
I think it would be useful if we have something like `os.path.join` in Python that would merge 2 parhs.

Reply:
Yes, especially if it's in common and can be used all over the codebase.